### PR TITLE
adding info on migration to shinytest2 

### PR DESCRIPTION
**Pull request by jen-machin**
Added in some info on migrating from shinytest to shinytest2, including potential issues people might run into. There were also some weird font formatting issues in the section above so I've changed the text format to fix them (I think the $ was being interpreted by markdown as something that modified the font).

### CODE REVIEW

File: writing-visualising/dashboards.qmd [hunk 1]
```diff
@@ -513,6 +513,39 @@ An example of a basic shiny test to check that your app loads up correctly can b
 
 ---
 
+#### Migration to shinytest2
+
+---
+
+The shinytest package has been superceded by the [shinytest2 package](https://rstudio.github.io/shinytest2/). If you already have shinytest UI tests in place for your dashboard, there is a function available to migrate these tests for you so that you do not have to rewrite them all yourself.
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:520]: Spelling - supersede rather than supercede, and I enjoy this mini description on google that I found when I questioned myself on it:

image

File: writing-visualising/dashboards.qmd [hunk 2]
```diff
@@ -502,8 +502,8 @@ Each test should produce one "expected results" JSON file. You can track particu
 For example, the [SCAP LA school places scorecards app](https://github.com/dfe-analytical-services/la-school-places-scorecards/blob/main/tests/shinytest/UI_tests.R) has tests that check that the app functions as expected when different combinations of filters are selected.
 
 Each test is separated out by comments, clearly stating what is being tested.
- - **app$setInputs()** tells the UI test to select an input in the app
- - **app$snapshot()** tells the UI test which outputs to capture (i.e. which outputs you'd expect to change and want to test)
+ - `app$setInputs()` tells the UI test to select an input in the app
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:505]: These bullet points aren't showing as expected, I think you might need an extra blank line before they start? [this branch]:
image

Though the main branch is even worse, so you've not broken it here [main branch that's live]:

image

File: writing-visualising/dashboards.qmd [hunk 3]
```diff
@@ -513,6 +513,39 @@ An example of a basic shiny test to check that your app loads up correctly can b
...
+
+**Chromote**
+
+Shinytest2 makes use of the Chromote package, which needs to connect to your local instance of Google Chrome. Sometimes your copy of Chrome might be installed in a different place to where Chromote expects to find it, which means that your tests will not be able to run. There is information on how to resolve this issue [in the Tips for Using R section](../learning-development/r.html#shinytest2---invalid-path-to-chrome-error-message)
+
+**Formatting of line endings**
+
+The migration function will only work if specific line endings are used in your code. If you receive an error that looks like this, then you may be having this issue:
+
+```
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:534]: I don't think this code block is formatting right? [image from local build on this branch]
image

File: writing-visualising/dashboards.qmd [hunk 4]
```diff
@@ -513,6 +513,39 @@ An example of a basic shiny test to check that your app loads up correctly can b
...
+
+There is [guidance on migrating to shinytest2](https://rstudio.github.io/shinytest2/articles/z-migration.html) available. To complete the migration, run `shinytest2::migrate_from_shinytest(PATH_TO_APP)`, making sure to include the full filepath. The function will search for your "tests" folder, and will then look inside the folder for any scripts containing tests. A new `testthat` folder will be automatically created, containing the code that runs your tests, as well as the snapshots generated when your tests are run. 
+
+There are a couple of potential issues that you may run into:
+
+**Chromote**
+
+Shinytest2 makes use of the Chromote package, which needs to connect to your local instance of Google Chrome. Sometimes your copy of Chrome might be installed in a different place to where Chromote expects to find it, which means that your tests will not be able to run. There is information on how to resolve this issue [in the Tips for Using R section](../learning-development/r.html#shinytest2---invalid-path-to-chrome-error-message)
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:528]: Scope creep a little - but in that section you refer to here we don't put anything on how to find where your chrome is installed. I wonder if someone has this issue, then it might be worth adding a quick paragraph on how to find that out? (I'm assuming there's a quick way like hit windows and right click chrome to find it or a command you could run or something to show you what path it should be?)

File: writing-visualising/dashboards.qmd [hunk 5]
```diff
@@ -513,6 +513,39 @@ An example of a basic shiny test to check that your app loads up correctly can b
...
+
+**Formatting of line endings**
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:530]: Do we need to reference this higher up - like 'many people may have X error message when trying to follow the shinytest2 guidance, see the section on line endings below for how to address this' or something like that?

File: writing-visualising/dashboards.qmd [hunk 6]
```diff
@@ -513,6 +513,39 @@ An example of a basic shiny test to check that your app loads up correctly can b
...
+20: app$snapshot()
+21: 
+    ^
+```
+
+You can modify the type of line endings used in your test code by going to `Tools > Global Options > Code` inside RStudio. In the "Saving" tab, there is a drop down box called "Line Ending Conversion". You need to change this to "Posix (LF)" as in the image below:
+
+![](../images/modify_line_endings.PNG)
+
+You should then save your test script, and re-run the `shinytest2::migrate_from_shinytest(PATH_TO_APP)` command.
```
Comment (cjrace) [inline writing-visualising/dashboards.qmd:545]: Should we explain what the PATH_TO_APP should be?

**Review by cjrace — CHANGES_REQUESTED** [review]

File: writing-visualising/dashboards.qmd (hunk 1 above)
Comment (jen-machin) [reply writing-visualising/dashboards.qmd:520]: I love this 😆
